ckan/lib/uploader.py

```diff
@@ -362,3 +362,29 @@ def download(self, id, filename=None):
             response.headers['Content-Type'] = content_type
         response.status = status
         return app_iter
+
+    def metadata(self, id, filename=None):
```

Comment:
Perhaps `download_metadata` would be a more intuitive name, to go with `download`?

Reply:
its file_metadata instead of download_metadata.
also need to add a action similar to package_show or download routing so it can be surfaced and called directly by rerouting in s3 instead of doubling up the logic inside archiver.